examples/src/main/scala/zio/config/examples/ProgramExample.scala

```diff
@@ -66,69 +35,65 @@ object ProgramExample extends App {
   }
 }
 
-// bootstrap modules
-object bootstrap {
-  type SparkSession = String
-
-  trait SparkEnv {
-    def spark: SparkEnv.Service
-  }
-
-  object SparkEnv {
-    type SparkSession = String
-
-    trait Service {
-      def getSpark: Task[SparkSession]
-    }
-  }
-
-  case class Bootstrap(configService: ConfigSource.Service, spark: SparkEnv.Service) extends SparkEnv with ConfigSource
+// spark service
+trait SparkEnv {
+  def spark: SparkEnv.Service
 }
 
-// client modules
-object client {
-
-  // Let sparksession be string for example simplicity.
+object SparkEnv {
   type SparkSession = String
 
-  case class Client(sparkSession: SparkSession, config: ProgramConfig) extends Spark with Config
-
-  trait Spark {
-    val sparkSession: SparkSession
+  trait Service {
+    def getSpark: UIO[SparkSession]
   }
 
-  trait Config {
-    val config: ProgramConfig
-  }
+  def make: IO[Throwable, SparkEnv] =
+    ZIO
+      .effect("sparkEnv")
+      .map(
+        r =>
+          new SparkEnv {
+            override def spark: Service = new Service {
+              override def getSpark: UIO[SparkSession] = ZIO.succeed(r)
+            }
+          }
+      )
 }
 
 // The core application
 object Application {
-  val processData: ZIO[client.Spark with client.Config, Throwable, Unit] =
-    ZIO.accessM(
-      r => ZIO.effect(println(s"Executing ${r.config.inputPath} and ${r.config.outputPath} using ${r.sparkSession}"))
-    )
+  val processData: ZIO[SparkEnv with Config[ProgramConfig], Throwable, Unit] =
+    ZIO.accessM(env => {
+      for {
+        conf  <- config[ProgramConfig]
+        spark <- env.spark.getSpark
+        _     <- ZIO.effect(println(s"Executing ${conf.inputPath} and ${conf.outputPath} using ${spark}"))
+      } yield ()
+    })
```

Comment:
you can define in  the object `SparkEnv`
```scala
 object Live extends Service {
  def getSpark: UIO[SparkSession] = ZIO.succeed(r)
}
```
then you will not need to call `make`, your program would be
```scala
 val pgm = for {
      confService  <- Config.fromEnv(programConfig)
      _            <- Application.execute.provide(Live(confService.config, SparkEnv.Live.spark))
    } yield ()
```
It would be better if you make `trait Live extends Config`and `trait Live extends SparkEnv` 
then you can provide the environment like this:
```scala
zio.provide(new Config.Live with SparkEn.Live)
```

Reply:
If I pull this to making spark as part of “trait Live” will I end up instantiating spark every time I access it?

Reply:
I could get the first part working, but Config.Live working. We can take this up in a follow up PR.

https://github.com/zio/zio-config/issues/59

Reply:
https://github.com/zio/zio-config/issues/46#issuecomment-537453426